tests/test_recipes.py

```diff
@@ -0,0 +1,108 @@
+import pytest
+
+import falcon
+import falcon.testing
+
+
+class TestMultipartMixed:
+    """Test parsing example from the now-obsolete RFC 1867:
+
+    --AaB03x
+    Content-Disposition: form-data; name="field1"
+
+    Joe Blow
+    --AaB03x
+    Content-Disposition: form-data; name="docs"
+    Content-Type: multipart/mixed; boundary=BbC04y
+
+    --BbC04y
+    Content-Disposition: attachment; filename="file1.txt"
+
+    This is file1.
+
+    --BbC04y
+    Content-Disposition: attachment; filename="file2.txt"
+
+    Hello, World!
+
+    --BbC04y--
+
+    --AaB03x--
+    """
+
+    @classmethod
+    def prepare_form(cls):
+        lines = [line.strip() for line in cls.__doc__.splitlines()[2:]]
+        return '\r\n'.join(lines).encode()
+
+    def test_parse(self, util):
+        recipe = util.load_module('examples/recipes/multipart_mixed_main.py')
+
+        result = falcon.testing.simulate_post(
+            recipe.app,
+            '/forms',
+            body=self.prepare_form(),
+            content_type='multipart/form-data; boundary=AaB03x',
+        )
+        assert result.status_code == 200
+        assert result.json == {
+            'file1.txt': 'This is file1.\r\n',
+            'file2.txt': 'Hello, World!\r\n',
+        }
+
+
+class TestOutputCSV:
+    @pytest.mark.parametrize(
+        'recipe,expected_head',
+        [
+            ('output_csv_text', '"fruit","quantity"\r\n"apples",13\r\n'),
+            ('output_csv_stream', '"n","Fibonacci Fn"\r\n0,0\r\n1,1\r\n'),
+        ],
+        ids=['simple', 'stream'],
+    )
+    def test_csv_output(self, asgi, util, recipe, expected_head):
+        suffix = 'asgi' if asgi else 'wsgi'
+        module = util.load_module(f'examples/recipes/{recipe}_{suffix}.py')
```

Comment:
maybe it would make sense to have a `util.load_recepie(recepie, asgi)` since a similar thing is used multiple times

Reply:
Done, added more affordances to load module to be able to easier parametrize.